### PR TITLE
Add a <rect> to Component background so it can be colored.

### LINKED ISSUE #1725: Add a rect element inside of render-area to allow styling background
Currently it's practically impossible to style the background content of a chart (inside the axis). Inside of .render-area we could add a rect element to make this much easier.

Below is the area we would like to have control over.

image

### PR DESCRIPTION
**Pull request by jtlan**
Select and color using:
```CSS
.plottable .background-fill {
  fill: pink;
}
```

Additional selectors can be inserted before "`.background-fill`" to narrow coloring to specific `Component`s.

Close #1725.

### CODE REVIEW

File: plottable.css [hunk 1]
```diff
@@ -43,6 +43,10 @@ svg.plottable {
   display : block; /* SVGs must be block elements for width/height calculations to work in Firefox. */
 }
 
+.plottable .background-fill {
```
Comment (bluong) [inline plottable.css]: We should probably think of what to name this background rectangle as it easily has more capability can just coloring the fill

File: src/components/abstractComponent.ts [hunk 2]
```diff
@@ -78,6 +78,11 @@ export module Component {
       this._cssClasses = null;
 
       this._backgroundContainer = this._element.append("g").classed("background-container", true);
+      this._boxes.push(
+        this._backgroundContainer.append("rect")
+                                 .classed("background-fill", true)
+                                 .attr("width", this.width()).attr("height", this.height())
```
Comment (bluong) [inline src/components/abstractComponent.ts]: Should we add a foreground rectangle?

Comment (bluong) [inline src/components/abstractComponent.ts]: Would using _addBox here work?

File: plottable.css (hunk 1 above)
Comment (jtlan) [inline plottable.css]: I guess? Affecting the stroke seems like a mistake though.

**Comment by bluong** [comment]
Just the naming is left.

**Comment by crmorford** [comment]
.background-fill sounds fine

**Comment by crmorford** [comment]
http://jsfiddle.net/n3swy1tf/

**Comment by jtlan** [comment]
Picture for the release:

screen shot 2015-03-12 at 8 47 36 pm

**Pull request merged**